native/effektio/src/api/messages.rs

```diff
@@ -33,6 +39,91 @@ impl RoomMessage {
     pub fn origin_server_ts(&self) -> u64 {
         self.inner.origin_server_ts.as_secs().into()
     }
+
+    pub fn msgtype(&self) -> String {
+        self.inner.content.msgtype().to_string()
+    }
+
+    pub async fn image_description(&self) -> Result<ImageDescription> {
+        match &self.inner.content.msgtype {
+            MessageType::Image(content) => {
+                let client = Client::current_client().unwrap();
+                let info = content.info.as_ref().unwrap();
+                RUNTIME.block_on(async move {
+                    let bin_data = client
+                        .get_media_content(
+                            &MediaRequest {
+                                source: content.source.clone(),
+                                format: MediaFormat::File,
+                            },
+                            false,
+                        )
+                        .await?;
+                    let description = ImageDescription {
+                        _bin_data: bin_data,
+                        _name: content.body.clone(),
+                        _mimetype: match info.mimetype.as_ref() {
+                            Some(value) => Some(value.clone()),
+                            None => None,
+                        },
+                        _size: match info.size {
+                            Some(value) => u64::from(value),
+                            None => 0,
+                        },
+                        _width: match info.width {
+                            Some(value) => Some(u64::from(value)),
+                            None => None,
+                        },
+                        _height: match info.height {
+                            Some(value) => Some(u64::from(value)),
+                            None => None,
+                        },
+                    };
+                    Ok(description)
+                })
+            }
+            _ => bail!("Invalid file format"),
+        }
+    }
+}
+
+pub struct ImageDescription {
+    _bin_data: Vec<u8>,
+    _name: String,
+    _mimetype: Option<String>,
+    _size: u64,
+    _width: Option<u64>,
+    _height: Option<u64>,
```

Comment:
in rust all fields are private, unless prefixed with `pub`. `_`-prefixes are only used for internal variables, that aren't actually needed. Let's stick to that model and stay with regular names instead.

```suggestion
    bin_data: Vec<u8>,
    name: String,
    mimetype: Option<String>,
    size: u64,
    width: Option<u64>,
    height: Option<u64>,
```

Reply:
Yes, rust recognize `_` as private property in struct.
By the way, `ImageDescription` must keep the getter functions about these fields.
So I appended the prefix `_` to properties.

Reply:
Sure, `_`-prefixes are acceptable, but are unnecessary (and make it needlessly harder to read), so I'd prefer we dropped them. Of course that doesn't mean we don't need the getter functions, that's totally okay. Was just talking about style...

Reply:
Understood.
Will change property name and keep getter function name.